cli/azd/docs/docgen.go

```diff
@@ -111,6 +116,21 @@ func addCodeFencesToSampleCommands(s string) string {
 	return strings.Join(newLines, "\n")
 }
 
+var precedingDollarRegexp = regexp.MustCompile(`^([\s]*)\$ (.*)$`)
+
+// Formats a command line by removing the dollar sign at the beginning of the
+// command while preserving any leading whitespace. line should be a string with
+// no newline "\n" characters.
+// Example:
+// `$ command --with someParameters` -> `command --with someParameters`
+// `  $ command --with someParameters` -> `  command --with someParameters`
+// No change if there are non-whitespace characters before the $
+// `  some string with a $ in the middle` -> `  some string with a $ in the middle`
+// `hello world` -> `hello world`
+func formatCommandLine(line string) string {
```

Comment:
You could consider adding a `docgen_test.go` and writing some tests for these if you wanted (more of a gentle nudge of "go makes it easy to write these tests", I think this is simple enough right now that we don't strictly need it, but it might be nice to have a test instead of a load bearing comment.